nmdc_server/ingest/pipeline.py

```diff
@@ -26,29 +26,25 @@ def __call__(self, db: Session, obj: Dict[str, Any], **kwargs: Any) -> LoadObjec
 
 # Load metagenome annotation as well as the gene function annotations produced.
 def load_mg_annotation(db: Session, obj: Dict[str, Any], **kwargs) -> LoadObjectReturn:
+    logger = get_logger(__name__)
     pipeline = schemas.MetagenomeAnnotationBase(**obj)
     row = models.MetagenomeAnnotation(**pipeline.dict())
     db.add(row)
     db.flush()
 
     annotations: Collection = kwargs["annotations"]
 
-    query = annotations.find(
-        {
-            "was_generated_by": pipeline.id,
-            "has_function": {
-                "$regex": ko_regex,
```

Comment:
I'm curious; now that the `was_generated_by` is indexed, does keeping the regex in the query slow it down materially? Right now we know it's not filtering many rows, but is there potential for it to be a stronger filter in the future?

Reply:
It's definitely a lot slower (not 4 hours, but 10+ min)

Reply:
Roger that. Did you try it with the regex but also including `.hint('was_generated_by')`? I think that's janky too, I'm just very curious.

(Don't feel the need to spend any more time on this if you don't want, BTW.)

Reply:
(The right solution is a compound index anyway.)

Reply:
I'm trying the hint.  I'd prefer that to the client side filter.

Reply:
No dice, merging.